### DISCUSSION TITLE
Dealing with unwanted service requests

### DISCUSSION THREAD

**Discussion by helo-head** [HomeSpan/HomeSpan#151] [Q&A]
I'm using the **door** service as part of my end solution as it includes all the characteristics I need. However my solution is a sensor only in that it provides current status information based on a the output of a set of accelerometers/gyros/ahrs but it cannot control the end device. 

This leaves me with the problem of dealing with client update requests. If a client accidentally tries to tell the door to open it will get stuck in opening state until the service reports back that the current position matches the target position. 

Is there any way to disallow updates of a service (maybe using locktargetstate or AdminOnlyAccess characteristics) or a way via an update handler to fool HomeKit into thinking the update was completed if only for a very short period ... For instance by grouping the update to say something like state = opening, state=stopped, position = open, target = closed, position = closed when the client requests the door be opened. 

Id use a sensor service to do the job but there is currently no sensor service that provides all the features Im looking for. Unfortunately even existing characteristics like LockTargetState and AdminOnlyAccess do not seem to be documented on Apples site or if so I cannot find them.

**Comment by HomeSpan** [comment]
Normally you would move the door to match targetState and then report the matching currentState back to HomeSpan.  But I don't see why the reverse would not work as well.  Within the update() method, try something like targetState->setVal(currentState->getVal()); 

This should (hopefully) cause the Home App to reset the target state to whatever the current state is, effectively  ignoring any changes the user makes in the Home App. And since the Home App will see both target and current states match, it should not report any movement of the door.

**Reply by helo-head** [reply]
Thanks, good idea. Unfortunately just resetting the target did not work. What does seem to work is to set the position to the requested target and then reset the target/position back to what it the correct value is. The down side to this solution is both updates trigger notifications .. Oh well, at least I have a short term solution.

Im going to play with the locktargetstate and AdminOnlyAccess to see if they do anything and what if any behavior is detectable. Although Apple seems to have a fair amount of documentation on Homekit there does seem to be a lot of holes in said documentation.
